src/lib/ssl_util.c

```diff
@@ -355,8 +344,15 @@ CK_RV ssl_util_encrypt(EVP_PKEY *pkey,
 
     size_t outlen = *ctextlen;
     rc = EVP_PKEY_encrypt(pkey_ctx, ctext, &outlen, ptext, ptextlen);
-    if (!rc) {
-        SSL_UTIL_LOGE("Could not perform RSA public encrypt");
```

Comment:
This patch exposes an issue.. where:
ERROR on line: "355" in file: "src/lib/ssl_util.c": Could not perform RSA public encrypt: error:0406D06E:rsa routines:RSA_padding_add_PKCS1_type_2:data too large for key size

EVP_PKEY_encrypt can return negative values, and they we're being treated as success.  I will have to dig into this more.